src/lib/mdns/Resolver_ImplMinimalMdns.cpp

```diff
@@ -35,6 +38,97 @@ namespace chip {
 namespace Mdns {
 namespace {
 
+enum class DiscoveryType
+{
+    UNKNOWN,
+    OPERATIONAL,
+    COMMISSIONABLE_NODE,
+};
+
+class TxtRecordDelegateImpl : public mdns::Minimal::TxtRecordDelegate
+{
+public:
+    TxtRecordDelegateImpl(CommissionableNodeData * nodeData) : mNodeData(nodeData) {}
+    void OnRecord(const mdns::Minimal::BytesRange & name, const mdns::Minimal::BytesRange & value);
+
+private:
+    CommissionableNodeData * mNodeData;
+};
+
+bool IsKey(const mdns::Minimal::BytesRange key, const char * desired)
+{
+    size_t len = strlen(desired);
+    if (key.Size() != len)
+    {
+        return false;
+    }
+    const char * keychars = reinterpret_cast<const char *>(key.Start());
+    for (size_t i = 0; i < len; ++i)
+    {
+        if (keychars[i] != desired[i])
+        {
+            return false;
+        }
+    }
+    return true;
+}
+
+uint16_t MakeU16(const mdns::Minimal::BytesRange & val)
```

Comment:
Does this match the spec on leading `0` handling?

Reply:
This doesn't have a strong opinion on leading zeros. So right now, if you send a leading zero, it will work fine. According to the spec as of 0.7 ballot 2, there aren't supposed to be leading zeros, but I'm also not keen on breaking device code based on that. Leading zeros in theory, should be handled by certification tests, but if one gets by, I would hope we could just roll with it.

Reply:
PR https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/2429 is removing all leading zeroes everywhere in discovery, since it adds no value.